src/mscorlib/corefx/System/Globalization/DateTimeFormatInfo.cs

```diff
@@ -2635,42 +2635,34 @@ private static bool IsHebrewChar(char ch)
                 // Check this value has the right category (regular token or separator token) that we are looking for.
                 if (((int)value.tokenType & (int)TokenMask) > 0 && value.tokenString.Length <= remaining)
                 {
-                    if (this.Culture.CompareInfo.Compare(str.Value, str.Index, value.tokenString.Length, value.tokenString, 0, value.tokenString.Length, CompareOptions.IgnoreCase) == 0)
+                    bool compareStrings = true;
+                    if (isLetter)
                     {
-                        if (isLetter)
+                        // If this token starts with a letter, make sure that we won't allow partial match.  So you can't tokenize "MarchWed" separately.
+                        // Also an optimization to avoid string comparison
+                        int nextCharIndex = str.Index + value.tokenString.Length;
+                        if (nextCharIndex > str.len)
                         {
-                            // If this token starts with a letter, make sure that we won't allow partial match.  So you can't tokenize "MarchWed" separately.
-                            int nextCharIndex;
-                            if ((nextCharIndex = str.Index + value.tokenString.Length) < str.len)
-                            {
-                                // Check word boundary.  The next character should NOT be a letter.
-                                char nextCh = str.Value[nextCharIndex];
-                                if (Char.IsLetter(nextCh))
-                                {
-                                    return (false);
-                                }
-                            }
+                            compareStrings = false;
                         }
+                        else if (nextCharIndex < str.len)
+                        {
+                            // Check word boundary.  The next character should NOT be a letter.
+                            char nextCh = str.Value[nextCharIndex];
+                            compareStrings = !(Char.IsLetter(nextCh));
```

Comment:
The check means the current token being looked at doesn't match the input string, so we need to look at the next token (we can't return here). However, the check for HasSpacesInMonthNames doesn't need to occur so I will change that.